src/lib/comms/sol-coap.c

```diff
@@ -293,17 +294,58 @@ sol_coap_uri_path_to_buf(const struct sol_str_slice path[],
         buflen--;
         cur++;
 
-        if (path[i].len >= buflen)
+        if (path[i].len > buflen)
             return cur;
 
         memcpy(buf + cur, path[i].data, path[i].len);
         cur += path[i].len;
-        buflen -= path[i].len;
     }
 
     return cur;
 }
 
+static char *
+packet_extract_path(const struct sol_coap_packet *req)
+{
+    const int max_count = 16;
+    struct sol_coap_option_value options[max_count];
+    struct sol_str_slice *path;
+    unsigned int i;
+    size_t path_len;
+    char *path_str;
+    int r;
+    uint16_t count;
+
+    SOL_NULL_CHECK(req, false);
+
+    r = coap_find_options(req, SOL_COAP_OPTION_URI_PATH, options, max_count);
+    SOL_INT_CHECK(r, < 0, NULL);
+    SOL_INT_CHECK(r, > max_count, NULL);
+    count = r;
+
+    path = alloca(sizeof(struct sol_str_slice) * (count + 1));
```

Comment:
I am checking that count is between 0 and 16. Is it dangerous anyway? I could use a fixed size of 16 (maximum) if so.